diff -Nru -- vscode.orig/src/vs/code/node/cli.ts vscode/src/vs/code/node/cli.ts
--- vscode.orig/src/vs/code/node/cli.ts	2026-09-18 17:29:56.958846833 +0200
+++ vscode/src/vs/code/node/cli.ts	2026-09-18 17:19:15.687205488 +0200
@@ -488,6 +488,7 @@
 		}
 
 		let child: ChildProcess;
+		const app_arg = '--app=' + process.env['ISABELLE_VSCODIUM_APP'];
 		if (!isMacintosh) {
 			if (!args.verbose && args.status) {
 				options['stdio'] = ['ignore', 'pipe', 'ignore']; // restore ability to see output when --status is used
@@ -499,7 +500,8 @@
 			const spawnArgs = isWindows ? combineUriFlags(argv.slice(2)) : argv.slice(2);
 
 			// We spawn the resolved executable directly
-			child = spawn(process.execPath, spawnArgs, options);
+			const electron_args = [app_arg].concat(spawnArgs);
+			child = spawn(process.execPath, electron_args, options);
 		} else {
 			// On macOS, we spawn using the open command to obtain behavior
 			// similar to if the app was launched from the dock
@@ -558,7 +560,8 @@
 				}
 			}
 
-			spawnArgs.push('--args', ...argv.slice(2)); // pass on our arguments
+			const electron_args = [app_arg].concat(...argv.slice(2));
+			spawnArgs.push('--args', ...electron_args); // pass on our arguments
 
 			if (env['VSCODE_DEV']) {
 				// If we're in development mode, replace the . arg with the
