diff -Nru -- vscode.orig/src/vs/code/node/cli.ts vscode/src/vs/code/node/cli.ts
--- vscode.orig/src/vs/code/node/cli.ts	2025-11-11 15:34:41.567132460 +0100
+++ vscode/src/vs/code/node/cli.ts	2025-11-11 15:34:42.202463807 +0100
@@ -482,13 +482,15 @@
 		}
 
 		let child: ChildProcess;
+		const app_arg = '--app=' + process.env['ISABELLE_VSCODIUM_APP'];
+		const electron_args = [app_arg].concat(argv.slice(2));
 		if (!isMacOSBigSurOrNewer) {
 			if (!args.verbose && args.status) {
 				options['stdio'] = ['ignore', 'pipe', 'ignore']; // restore ability to see output when --status is used
 			}
 
 			// We spawn process.execPath directly
-			child = spawn(process.execPath, argv.slice(2), options);
+			child = spawn(process.execPath, electron_args, options);
 		} else {
 			// On Big Sur, we spawn using the open command to obtain behavior
 			// similar to if the app was launched from the dock
@@ -548,7 +550,7 @@
 				}
 			}
 
-			spawnArgs.push('--args', ...argv.slice(2)); // pass on our arguments
+			spawnArgs.push('--args', ...electron_args); // pass on our arguments
 
 			if (env['VSCODE_DEV']) {
 				// If we're in development mode, replace the . arg with the
