diff --git a/extensions/theme-seti/build/update-icon-theme.js b/extensions/theme-seti/build/update-icon-theme.js
index 366e7f3..0305e0f 100644
--- a/extensions/theme-seti/build/update-icon-theme.js
+++ b/extensions/theme-seti/build/update-icon-theme.js
@@ -29,7 +29,6 @@ const nonBuiltInLanguages = { // { fileNames, extensions  }
 	"kotlin": { extensions: ['kt'] },
 	"mustache": { extensions: ['mustache', 'mst', 'mu', 'stache'] },
 	"nunjucks": { extensions: ['nunjucks', 'nunjs', 'nunj', 'nj', 'njk', 'tmpl', 'tpl'] },
-	"ocaml": { extensions: ['ml', 'mli', 'mll', 'mly', 'eliom', 'eliomi'] },
 	"puppet": { extensions: ['puppet'] },
 	"r": { extensions: ['r', 'rhistory', 'rprofile', 'rt'] },
 	"rescript": { extensions: ['res', 'resi'] },
