diff -Nru -- vscode.orig/extensions/theme-seti/build/update-icon-theme.js vscode/extensions/theme-seti/build/update-icon-theme.js
--- vscode.orig/extensions/theme-seti/build/update-icon-theme.js	2025-11-11 15:34:41.468787385 +0100
+++ vscode/extensions/theme-seti/build/update-icon-theme.js	2025-11-11 15:34:42.220516055 +0100
@@ -29,7 +29,6 @@
 	"kotlin": { extensions: ['kt'] },
 	"mustache": { extensions: ['mustache', 'mst', 'mu', 'stache'] },
 	"nunjucks": { extensions: ['nunjucks', 'nunjs', 'nunj', 'nj', 'njk', 'tmpl', 'tpl'] },
-	"ocaml": { extensions: ['ml', 'mli', 'mll', 'mly', 'eliom', 'eliomi'] },
 	"puppet": { extensions: ['puppet'] },
 	"r": { extensions: ['r', 'rhistory', 'rprofile', 'rt'] },
 	"rescript": { extensions: ['res', 'resi'] },
