diff --git a/prepare_vscode.sh b/prepare_vscode.sh
index 1ac30e0..66efd47 100755
--- a/prepare_vscode.sh
+++ b/prepare_vscode.sh
@@ -225,13 +225,16 @@ for i in {1..5}; do # try 5 times
   sleep $(( 15 * (i + 1)))
 done
 
+# see https://github.com/electron/electron/blob/v42.8.1/docs/breaking-changes.md
+npx install-electron --no
+
 mv .npmrc.bak .npmrc
 # }}}
 
 # package.json
 cp package.json{,.bak}
 
-setpath "package" "version" "${RELEASE_VERSION%-insider}"
+setpath "package" "version" "$MS_TAG"
 
 replace 's|Microsoft Corporation|VSCodium|' package.json
 replace "s|--max-old-space-size=8192|--max-old-space-size=${MAX_OLD_SPACE_SIZE}|" package.json
