diff --git a/icons/build_icons.sh b/icons/build_icons.sh
index bca6852..d303f6f 100755
--- a/icons/build_icons.sh
+++ b/icons/build_icons.sh
@@ -26,24 +26,51 @@ check_programs() { # {{{
   for arg in "$@"; do
     if ! command -v "${arg}" &> /dev/null; then
       echo "${arg} could not be found"
-      exit 0
+      exit 1
     fi
   done
 } # }}}
 
-check_programs "icns2png" "composite" "convert" "png2icns" "icotool" "rsvg-convert" "sed"
+check_programs "composite" "convert" "rsvg-convert" "sed"
 
 . "./${VSCODE_PREFIX}utils.sh"
 
+# Generate icons according to https://github.com/VSCodium/icons
+ICONS_ROOT="https://raw.githubusercontent.com/VSCodium/icons/b96d266673cd1d7a09d4ddd869c669fcb7a1e8ba"
+BG_PATH="./${VSCODE_PREFIX}icons"
+BG_FILENAME="main.svg"
+cp "$BG_PATH/${QUALITY}/codium_cnl.svg" "icon_head.svg"
+
 if ! declare -F load_linux_png &>/dev/null; then
   load_linux_png() {
-    wget "https://raw.githubusercontent.com/VSCodium/icons/main/icons/linux/circle1/${COLOR}/paulo22s.png" -O "$1"
+    curl -fsSL "$ICONS_ROOT/templates/backgrounds/circle1/main.svg" -o "$BG_PATH/$BG_FILENAME"
+    curl -fsSL "$ICONS_ROOT/templates/backgrounds/circle1/main.conf" -o "main.conf"
+    source "main.conf"
+
+    convert -size 2048x2048 canvas:transparent PNG32:"icon.png"
+    rsvg-convert -w 2048 -h 2048 "${BG_PATH}/${BG_FILENAME}" -o "icon_bg.png"
+    composite "icon_bg.png" -gravity center -background none -colorspace sRGB "icon.png" "icon.png"
+    rsvg-convert -w "${BG_SHAPE_SIZE}" -h "${BG_SHAPE_SIZE}" "icon_head.svg" -o "icon_head.png"
+    composite "icon_head.png" -geometry "+${BG_SHAPE_LEFT}+${BG_SHAPE_TOP}" -background none -colorspace sRGB "icon.png" "icon.png"
+
+    composite \( "icon.png" -resize "${BG_LINUX_SIZE}x${BG_LINUX_SIZE}" \) -gravity center -background none \( -size 1024x1024 canvas:transparent \) "$1"
+
+    rm icon.png icon_bg.png icon_head.png
   }
 fi
 
 if ! declare -F load_windows_ico &>/dev/null; then
   load_windows_ico() {
-    wget "https://raw.githubusercontent.com/VSCodium/icons/main/icons/win32/nobg/${COLOR}/paulo22s.ico" -O "$1"
+    curl -fsSL "$ICONS_ROOT/templates/backgrounds/nobg/main.conf" -o "main.conf"
+    source "main.conf"
+
+    convert -size 2048x2048 canvas:transparent PNG32:"icon.png"
+    rsvg-convert -w "${BG_SHAPE_SIZE}" -h "${BG_SHAPE_SIZE}" "icon_head.svg" -o "icon_head.png"
+    composite "icon_head.png" -gravity center -background none -colorspace sRGB "icon.png" "icon.png"
+
+    convert "icon.png" -define icon:auto-resize=256,128,96,64,48,32,24,20,16 "$1"
+
+    rm icon.png icon_head.png
   }
 fi
 
diff --git a/prepare_vscode.sh b/prepare_vscode.sh
index 1ac30e0..5e8ef9e 100755
--- a/prepare_vscode.sh
+++ b/prepare_vscode.sh
@@ -93,18 +93,18 @@ if [[ "${VSCODE_QUALITY}" == "insider" ]]; then
   setpath "product" "win32ContextMenu.x64.clsid" "90AAD229-85FD-43A3-B82D-8598A88829CF"
   setpath "product" "win32ContextMenu.arm64.clsid" "7544C31C-BDBF-4DDF-B15E-F73A46D6723D"
 else
-  setpath "product" "nameShort" "VSCodium"
-  setpath "product" "nameLong" "VSCodium"
-  setpath "product" "applicationName" "codium"
-  setpath "product" "linuxIconName" "vscodium"
+  setpath "product" "nameShort" "Isacode"
+  setpath "product" "nameLong" "Isabelle VSCode"
+  setpath "product" "applicationName" "isacode"
+  setpath "product" "linuxIconName" "isacode"
   setpath "product" "quality" "stable"
-  setpath "product" "urlProtocol" "vscodium"
+  setpath "product" "urlProtocol" "isacode"
   setpath "product" "serverApplicationName" "codium-server"
   setpath "product" "serverDataFolderName" ".vscodium-server"
-  setpath "product" "darwinBundleIdentifier" "com.vscodium"
-  setpath "product" "win32AppUserModelId" "VSCodium.VSCodium"
+  setpath "product" "darwinBundleIdentifier" "isabelle.vscode"
+  setpath "product" "win32AppUserModelId" "Isabelle.VSCode"
   setpath "product" "win32DirName" "VSCodium"
-  setpath "product" "win32MutexName" "vscodium"
+  setpath "product" "win32MutexName" "isacode"
   setpath "product" "win32NameVersion" "VSCodium"
   setpath "product" "win32RegValueName" "VSCodium"
   setpath "product" "win32ShellNameShort" "VSCodium"
@@ -225,15 +225,18 @@ for i in {1..5}; do # try 5 times
   sleep $(( 15 * (i + 1)))
 done
 
+# see https://github.com/electron/electron/blob/v42.8.1/docs/breaking-changes.md
+npx install-electron --no
+
 mv .npmrc.bak .npmrc
 # }}}
 
 # package.json
 cp package.json{,.bak}
 
-setpath "package" "version" "${RELEASE_VERSION%-insider}"
+setpath "package" "version" "$MS_TAG"
 
-replace 's|Microsoft Corporation|VSCodium|' package.json
+replace 's|Microsoft Corporation|Isabelle Prover Organisation|' package.json
 replace "s|--max-old-space-size=8192|--max-old-space-size=${MAX_OLD_SPACE_SIZE}|" package.json
 
 cp resources/server/manifest.json{,.bak}
@@ -251,8 +254,8 @@ replace "s|\\[\\/\\* BUILTIN_ANNOUNCEMENTS \\*\\/\\]|$( tr -d '\n' < ../announce
 
 ../undo_telemetry.sh
 
-replace 's|Microsoft Corporation|VSCodium|' build/lib/electron.ts
-replace 's|([0-9]) Microsoft|\1 VSCodium|' build/lib/electron.ts
+replace 's|Microsoft Corporation|Isabelle Prover Organisation|' build/lib/electron.ts
+replace 's|([0-9]) Microsoft|\1 Isabelle Prover|' build/lib/electron.ts
 
 if [[ "${OS_NAME}" == "linux" ]]; then
   # microsoft adds their apt repo to sources
diff --git a/utils.sh b/utils.sh
index 132812c..09ce3e4 100755
--- a/utils.sh
+++ b/utils.sh
@@ -1,12 +1,12 @@
 #!/usr/bin/env bash
 
-APP_NAME="${APP_NAME:-VSCodium}"
+APP_NAME="${APP_NAME:-Isacode}"
 APP_NAME_LC="$( echo "${APP_NAME}" | awk '{print tolower($0)}' )"
 ASSETS_REPOSITORY="${ASSETS_REPOSITORY:-VSCodium/vscodium}"
-BINARY_NAME="${BINARY_NAME:-codium}"
+BINARY_NAME="${BINARY_NAME:-isacode}"
 GH_REPO_PATH="${GH_REPO_PATH:-VSCodium/vscodium}"
-ORG_NAME="${ORG_NAME:-VSCodium}"
-TUNNEL_APP_NAME="${TUNNEL_APP_NAME:-"${BINARY_NAME}-tunnel"}"
+ORG_NAME="${ORG_NAME:-Isabelle Prover Organisation}"
+TUNNEL_APP_NAME="${TUNNEL_APP_NAME:-"codium-tunnel"}"
 
 if [[ "${VSCODE_QUALITY}" == "insider" ]]; then
   GLOBAL_DIRNAME="${GLOBAL_DIRNAME:-"${APP_NAME_LC}"}-insiders"
