--- a/generic/proof-site.el	2008-07-23 14:40:14.000000000 +0200
+++ b/generic/proof-site.el	2009-11-28 16:13:56.409505412 +0100
@@ -55,7 +55,7 @@
 
 (eval-and-compile
 ;; WARNING: do not edit next line (constant is edited in Makefile.devel)
-  (defconst proof-general-version "Proof General Version 3.7.1. Released by da on Wed 23 Jul 2008."
+  (defconst proof-general-version "Proof General Version 3.7.1.1. Fabricated by makarius on Sat 28 Nov 2009."
     "Version string identifying Proof General release."))
 
 (defconst proof-general-short-version 
@@ -64,7 +64,7 @@
       (string-match "Version \\([^ ]+\\)\\." proof-general-version)
       (match-string 1 proof-general-version))))
 
-(defconst proof-general-version-year "2008")
+(defconst proof-general-version-year "2009")
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;
